packages/block-library/src/post-date/edit.js

```diff
@@ -33,47 +33,30 @@ export default function PostDateEdit( {
 	context: { postId, postType, queryId },
 	setAttributes,
 } ) {
+	const blockProps = useBlockProps( {
+		className: classnames( {
+			[ `has-text-align-${ textAlign }` ]: textAlign,
+		} ),
+	} );
+	const timeRef = useRef();
 	const isDescendentOfQueryLoop = Number.isFinite( queryId );
 	const [ siteFormat ] = useEntityProp( 'root', 'site', 'date_format' );
+	const [ siteTimeFormat ] = useEntityProp( 'root', 'site', 'time_format' );
 	const [ date, setDate ] = useEntityProp(
 		'postType',
 		postType,
 		'date',
 		postId
 	);
-	const settings = __experimentalGetSettings();
-	// To know if the current time format is a 12 hour time, look for "a".
-	// Also make sure this "a" is not escaped by a "/".
-	const is12Hour = /a(?!\\)/i.test(
-		settings.formats.time
-			.toLowerCase() // Test only for the lower case "a".
-			.replace( /\\\\/g, '' ) // Replace "//" with empty strings.
-			.split( '' )
-			.reverse()
-			.join( '' ) // Reverse the string and test for "a" not followed by a slash.
-	);
-	const formatOptions = Object.values( settings.formats ).map(
-		( formatOption ) => ( {
-			key: formatOption,
-			name: dateI18n( formatOption, date ),
-		} )
-	);
-	const resolvedFormat = format || siteFormat || settings.formats.date;
-	const blockProps = useBlockProps( {
-		className: classnames( {
-			[ `has-text-align-${ textAlign }` ]: textAlign,
-		} ),
-	} );
-
-	const timeRef = useRef();
 
 	let postDate = date ? (
 		<time dateTime={ dateI18n( 'c', date ) } ref={ timeRef }>
-			{ dateI18n( resolvedFormat, date ) }
+			{ dateI18n( format || siteFormat, date ) }
```

Comment:
I think we might need a hook somewhere to ensure `siteFormat` is available.

For me at least, there seems to be a race condition whereby, if I refresh the Post Editor with a Post Date block and the site default selected, both `format` and `siteFormat` are `undefined`. 

This is only in the Post Editor. The Site Editor works as expected.

The result is, when refreshing the page, [format() complains that it cannot find a length](https://github.com/WordPress/gutenberg/blob/d5915916abc45e6682f4bdb70888aa41e98aa395/packages/date/src/index.js#L387).

Adding a new Post Date block reveals that the value is indeed returned at some point after refresh.

Reply:
Ah bugger. I guess it happens while the site entity is loading.

I added a fallback to the `@wordpress/date` format in b3fcfdaa3a114b1c973a15d45952cea71c0bfa3c.